apiary.apib

```diff
@@ -542,7 +542,186 @@ API endpoints that Operation Code's Rails backend makes available to its React f
             errors: "Some error message"
         }
 
+## Mentor Request | Create a New Mentor Request [/api/v1/requests{?user_id,details,language,service_id,requested_mentor_id,status,assigned_mentor_id}]
++ Parameters
+    
+    + user_id (integer, required) - Identification of the User requesting a mentor in the form of an integer
+    + details (string, optional) - Details of the mentor request in the form of a string
+    + language (string, optional) - Primary language of the requesting mentee in the form of a string
+    + service_id (integer, optional) - Indentification of the mentorship request in the form of an integer
+    + requested_mentor_id (integer, optional) - Requested mentor's id in the form of an integer
+    + status (string, optional) - Current fullfillment status in the form on a string
+    + assigned_mentor_id (integer, optional) - Assigned mentor'd id in the form of an integer
+    
+    
+### Create a New Mentor Request [POST]
++ Request (application/json)
+
+    + Headers
+    
+            Authorization: Bearer Access-Token
+
+    + Body
+
+            {
+                "user_id": 4600,
+                "details": "I want to type well",
+                "language": "English",
+                "service_id": 200,
+                "requested_mentor_id": 250,
+                "status": "Open",
+                "assigned_mentor_id": 250
+            }
+
++ Response 201 (application/json)
+
+    + Body
+
+            {   
+                "id": 2130,
+                "user_id": 4600,
+                "details": "I want to type well",
+                "language": "English",
+                "service_id": 200,
+                "requested_mentor_id": 250,
+                "status": "Open",
+                "assigned_mentor_id": 250
+            }
+
++ Response 422 (application/json)
```

Comment:
The [create action](https://github.com/OperationCode/operationcode_backend/blob/master/app/controllers/api/v1/requests_controller.rb#L11-L14) is current **not** accounting for an error response.  

Here is [an example](https://github.com/OperationCode/operationcode_backend/blob/master/app/controllers/api/v1/team_members_controller.rb#L15) of a different controller method that **is** accounting for error handling.


As such, you can adjust this whole section to return a `401`, like you did in the `index` action.